src/Linux/Native/RunAsUser/CMakeList.txt

```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 2.7)
+project(RunAsUser)
+
+if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+    message(FATAL_ERROR "In-source builds are not allowed. Please use the ./build.sh helper script.")
+endif()
+
+if(NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_HOME_DIRECTORY}/bin/${CMAKE_BUILD_TYPE})
+endif()
+
+# TODO: add -fstack-protector
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -fpermissive")
+
+file(GLOB src "src/*.h" "src/*.cpp" )
+
+add_executable(Microsoft.R.Host.RunAsUser ${src})
+
+set_target_properties(Microsoft.R.Host.RunAsUser PROPERTIES LINK_FLAGS "-static-libgcc -static-libstdc++")
+if("${TARGET_ARCH}" STREQUAL "x86")
+    set_target_properties(Microsoft.R.Host.RunAsUser PROPERTIES COMPILE_FLAGS "-m32" LINK_FLAGS "-m32")
+endif()
+
+include_directories("${CMAKE_SOURCE_DIR}/../../Lib/picojson")
+
+set(Boost_USE_STATIC_LIBS ON)
+find_package(Boost 1.58.0 REQUIRED COMPONENTS filesystem system)
+include_directories(${Boost_INCLUDE_DIRS})
+target_link_libraries(Microsoft.R.Host.RunAsUser ${Boost_LIBRARIES})
+
+target_link_libraries(Microsoft.R.Host.RunAsUser pthread pam explain)
```

Comment:
I believe you need to use `find_library` or `find_package` for `pam` and `explain` for this to be properly cross-plat. If `find_package` supports them, that one is preferable.